### PR TITLE
backend implemented

### LINKED ISSUE #1: Backend Implementation
The todo application should have a backend in NodeJS.
- Todo module is needed.
- Authentication module is needed via JWT.
- MongoDB should be used as Database.

### PR DESCRIPTION
**Pull request by dijitalYoruk**
Backend implemented in NodeJS with reasonable routers.

### CODE REVIEW

File: backend/src/setupTesting.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+   const payload = {
+      username: 'testUser',
+      email: 'test@test.com',
+      id: new mongoose.Types.ObjectId().toHexString(),
+   };
+
+   // Create the JWT!
+   const token = jwt.sign(payload, process.env.JWT_SECRET!);
+   return `Bearer ${token}`;
+};
```
Comment (ErdemTuna2) [inline backend/src/setupTesting.ts]: No need for this global method. In the tests, sign in  is already specified.

Comment (ErdemTuna2) [inline backend/src/setupTesting.ts]: No need for this code. Sign In is already handled in the actual tests.

Comment (dijitalYoruk) [reply backend/src/setupTesting.ts]: Yes you are right I will fix this.

**Comment by ErdemTuna2** [comment]
Looks fine. You can merge it.

**Pull request merged**